File: marketmotormicroservices/producto-service/src/main/java/com/jma/productoservice/detalleVenta/domain/response/DetalleVentaResponse.java
package com.jma.productoservice.detalleVenta.domain.response;

import com.jma.productoservice.detalleVenta.domain.dto.DetalleVentaDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DetalleVentaResponse {

    private List<DetalleVentaDto> content;
    private int pageNo;
    private int pageSize;
    private long totalElements;
    private int totalPages;
    private boolean last;

}

File: frontend-marketmotor/src/app/models/responseapi/DetalleVentaResponse.ts
import { DetalleOrdenCompra } from "../dtos/DetalleOrdenCompra";
import { DetalleVenta } from "../dtos/DetalleVenta";

export class DetalleVentaResponse{

    content: DetalleVenta[];
    pageNo: number;
    pageSize: number;
    totalElements: number;
    totalPages: number;
    last: boolean;

    constructor(){
        this.content  = [new DetalleVenta()]
        this.pageNo = 0;
        this.pageSize = 0;
        this.totalElements = 0;
        this.totalPages = 0;
        this.last = false;
    }

}